lib/benchmark/linear_test.go

```diff
@@ -0,0 +1,211 @@
+package benchmark
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerate(t *testing.T) {
+	d, _ := time.ParseDuration("30s")
+
+	initial := Config{
+		Rate:                0,
+		Command:             []string{"ls"},
+		Interactive:         false,
+		MinimumWindow:       d,
+		MinimumMeasurements: 1000,
+	}
+
+	linearConfig := Linear{
+		LowerBound:          10,
+		UpperBound:          50,
+		Step:                10,
+		MinimumMeasurements: 1000,
+		MinimumWindow:       d,
+		config:              initial,
+	}
+	// First generation
+	res := linearConfig.Generate()
+	if res != true {
+		t.Errorf("failed to generate first generation")
+	}
+	bm, err := linearConfig.GetBenchmark()
+	if err != nil {
+		t.Errorf("failed to get current benchmark")
+	}
+	expected := initial
+	expected.Rate = 10
+
+	assert.Empty(t, cmp.Diff(expected, bm))
+
+	// Second generation
+	res = linearConfig.Generate()
+	if res != true {
+		t.Errorf("failed to generate #2 generation")
+	}
+
+	bm, err = linearConfig.GetBenchmark()
+	if err != nil {
+		t.Errorf("failed to get current benchmark")
+	}
+
+	expected.Rate = 20
+	assert.Empty(t, cmp.Diff(expected, bm))
+
+	// Third generation
+	res = linearConfig.Generate()
+	if res != true {
+		t.Errorf("failed to generate #3 generation")
+	}
+
+	bm, err = linearConfig.GetBenchmark()
+	if err != nil {
+		t.Errorf("failed to get current benchmark")
+	}
+	expected.Rate = 30
+	assert.Empty(t, cmp.Diff(expected, bm))
+
+	// Fourth generation
+	res = linearConfig.Generate()
+	if res != true {
+		t.Errorf("failed to generate #4 generation")
+	}
+
+	bm, err = linearConfig.GetBenchmark()
+	if err != nil {
+		t.Errorf("failed to get current benchmark")
+	}
+	expected.Rate = 40
+	assert.Empty(t, cmp.Diff(expected, bm))
+
+	// Fifth generation
+	res = linearConfig.Generate()
+	if res != true {
+		t.Errorf("failed to generate #5 generation")
+	}
+
+	bm, err = linearConfig.GetBenchmark()
+	if err != nil {
+		t.Errorf("failed to get current benchmark")
+	}
+	expected.Rate = 50
+	assert.Empty(t, cmp.Diff(expected, bm))
+
+	// Sixth generation, should return false
+	res = linearConfig.Generate()
+	if res != false {
+		t.Errorf("expected false, got true")
+	}
+
+	_, err = linearConfig.GetBenchmark()
+	if err == nil {
+		t.Errorf("generating more benchmarks than expected")
+	}
+}
+
+func TestGenerateNotEvenMultiple(t *testing.T) {
```

Comment:
Same here - could you use table driven tests for this too?

https://github.com/golang/go/wiki/TableDrivenTests